packages/dd-trace/src/config.js

```diff
@@ -998,8 +997,7 @@ class Config {
     this._setBoolean(opts, 'runtimeMetricsRuntimeId', options.runtimeMetricsRuntimeId)
     this._setArray(opts, 'sampler.spanSamplingRules', reformatSpanSamplingRules(options.spanSamplingRules))
     this._setUnit(opts, 'sampleRate', options.sampleRate ?? options.ingestion.sampleRate)
-    const ingestion = options.ingestion || {}
-    opts['sampler.rateLimit'] = coalesce(options.rateLimit, ingestion.rateLimit)
+    opts['sampler.rateLimit'] = maybeInt(options.rateLimit ?? options.ingestion.rateLimit)
```

Comment:
needs optional chaining to be iso with the previous code no ?
```suggestion
    opts['sampler.rateLimit'] = maybeInt(options.rateLimit ?? options.ingestion?.rateLimit)
```